src/libraries/System.IO.FileSystem/src/System/IO/EnumerationOptions.cs

```diff
@@ -96,6 +97,12 @@ internal static EnumerationOptions FromSearchOption(SearchOption searchOption)
         /// </remarks>
         public MatchCasing MatchCasing { get; set; }
 
+        /// <summary>Gets or sets a value that indicates the maximum directory depth to recurse while enumerating, when <see cref="RecurseSubdirectories" /> is set to <see langword="true" />.</summary>
+        /// <value>A number that represents the maximum directory depth to recurse while enumerating. The default value is <see cref="int.MaxValue" />.</value>
+        /// <remarks>If <see cref="MaxRecursionDepth" /> is set to a negative number, the default value <see cref="int.MaxValue" /> is used.
+        /// If <see cref="MaxRecursionDepth" /> is set to zero, enumeration returns the contents of the initial directory.</remarks>
+        public int MaxRecursionDepth { get; set; }
```

Comment:
> indicates the maximum directory depth to recurse while enumerating, when <see cref="RecurseSubdirectories" /> is set to <see langword="true"

Can we consider throw when someone tries to set a value to this property and `RecurseSubdirectories` is false? It would make more clear that `MaxRecursionDepth` is not honored in that case.

Reply:
In my opinion, it would make sense if we did not have the default, but we have regardless of the value of `RecurseSubdirectories`.

Reply:
I can think of a few options to deal with this:

* Keep MaxRecursionDepth as int.MaxValue and throw when set_MaxRecursionDepth is called if RecurseSubdirectories == false (what I already proposed in my previous comment).
* Keep MaxRecursionDepth as default (zero) and change it to int.MaxValue when RecurseSubdirectories is set to true. Then we can throw when someone tries to call set_MaxRecursionDepth if RecurseSubdirectories == false (I feel that is not very intuitive that a property changes if another one does).
* Do nothing and expect that the users have read the documentation so they can know that MaxRecursionDepth doesn't work on its own and that is only honored when RecurseSubdirectories == true (what we already have).


cc @carlossanlop @adamsitnik

Reply:
I prefer third case. The API allows flipping RecurseSubdirectories on the fly so we would have very complex check code. Recursion is common scenario, checking depth is temporary one. If users turn off recursion and set MaxRecursionDepth to a value, they will not get a destructive result - again, there is always a default value, which also implies that there is nothing destructive there.